File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/dogecommander/bot/Bot.java
package org.firstinspires.ftc.teamcode.dogecommander.bot;

import com.disnodeteam.dogecommander.DogeBot;
import com.qualcomm.robotcore.hardware.HardwareMap;

import org.firstinspires.ftc.teamcode.dogecommander.bot.subsystems.Drive;
import org.firstinspires.ftc.teamcode.dogecommander.bot.subsystems.Intake;

public class Bot extends DogeBot {
    public Drive drive;
    public Intake intake;

    public Bot(HardwareMap hardwareMap) {
        drive = new Drive(hardwareMap);
        intake = new Intake(hardwareMap);

        addSubsystem(drive);
        addSubsystem(intake);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/bot/Bot.kt
package org.firstinspires.ftc.teamcode.bot

import com.disnodeteam.dogecommander.DogeBot
import com.disnodeteam.dogecommander.Subsystem
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.bot.subsystems.MecanumDrive

class Bot(hardwareMap: HardwareMap) : DogeBot() {
    val drive: MecanumDrive by lazy {MecanumDrive(hardwareMap)}

    override fun init() {
        addSubsystem(drive)
    }

    private operator fun Subsystem.unaryPlus() {
        addSubsystem(this)
    }
}